src/sdk/namespace_fs.js

```diff
@@ -545,6 +554,11 @@ class NamespaceFS {
         let buffer_pool_cleanup = null;
         const fs_context = this.prepare_fs_context(object_sdk);
         let file_path;
+
+        if (this.versioning === 'ENABLED' || this.versioning === 'SUSPENDED') {
```

Comment:
this is a bit hacky, I think it's preferred to have here the same behavior as in read_object_md:
get the version path and handle delete marker and version id etc

Reply:
Since we already have object md, I felt we can use the whatever information we got from md call which does save some IO operations. However I have replaced `get_file_path()` with `get_version_path()`, May be we can think of optimization here to avoid multiple IO operations

Reply:
How did it even pass test_namespace_fs.js tests? 
test_namespace_fs calls directly to the namespace runtime object and it's not going through the s3_put_object.js flow.
Optimization is always good and I guess we can optimize the 2 calls to get_version_path() to 1 but we need to take into consideration the scale and the fact that the file can be changed between the read_object_stream and read_object_md